src/review-todo.js

```diff
@@ -0,0 +1,212 @@
+let isDate = require("date-fns/isDate");
+
+function todo(todoName = "New Todo") {
+	let title = todoName,
+		description = "",
+		urgency = 0,
+		isDone = false,
+		dueDate = todoDate(),
+		scheduleDate = todoDate(),
+		tags = tagList();
+
+	let todoObj = Object.create(todo.proto);
+	Object.assign(todoObj, {
+		title,
+		description,
+		isDone,
+		urgency,
+		dueDate,
+		scheduleDate,
+		tags,
+	});
+
+	return todoObj;
+}
+
+todo.proto = {};
```

Comment:
I was trying to find ways to make the code extensible. Glad you pointed this out because I don't know how useful it is to do it this way, if I'm using constructor functions F.prototype does the same thing though I think.

Reply:
I'm a big fan of treating every line of code added as a liability and not taking on that liability without a clear benefit. The opposite runs the risk of "premature optimization". This extensibility can be added really easily if it's needed down the line, I think.